### PR TITLE
fix: optional orgs can be undefined

### PR DESCRIPTION
**Pull request by mshanemc**
@W-10967216@

### CODE REVIEW

File: src/flags/orgFlags.ts [hunk 1]
```diff
@@ -10,7 +10,14 @@ import { Messages, Org, ConfigAggregator } from '@salesforce/core';
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/sf-plugins-core', 'messages');
 
-const maybeGetOrg = async (input?: string): Promise<Org | undefined> => Org.create({ aliasOrUsername: input });
+const maybeGetOrg = async (input?: string): Promise<Org | undefined> => {
```
Comment (peternhale) [inline src/flags/orgFlags.ts:13]: How about some tests for all the org flags?

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/flags/orgFlags.ts [hunk 2]
```diff
@@ -10,7 +10,14 @@ import { Messages, Org, ConfigAggregator } from '@salesforce/core';
...
+  try {
+    return Org.create({ aliasOrUsername: input });
+  } catch (e) {
```
Comment (peternhale) [inline src/flags/orgFlags.ts:16]: The catch should be checking to see if the user provided a value. If so, the the error should be re-thrown.

File: src/flags/orgFlags.ts (hunk 1 above)
Comment (peternhale) [inline src/flags/orgFlags.ts:13]: My use case for this flag is to specify a prod org, which I believe is another category of of a well known org. How does one bind a different config value to the flag?

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/flags/orgFlags.ts (hunk 1 above)
Comment (mshanemc) [reply src/flags/orgFlags.ts:13]: I tried a few ways and hated all of them.  Open to suggestions or contributions

**Pull request merged**